NICE.Identity.Management/ClientApp/src/views/ServicesList/ServicesList.tsx

```diff
@@ -0,0 +1,411 @@
+
+import { fetchData } from "../../helpers/fetchData";
+import { isDataError } from "../../helpers/isDataError";
+import { Endpoints } from "../../data/endpoints";
+import { HistoryType, WebsiteType } from "../../models/types";
+import { FilterSearch } from "../../components/FilterSearch/FilterSearch";
+import { FilterBox } from "../../components/FilterBox/FilterBox";
+import { WebsiteEnvironment } from "../../components/WebsiteEnvironment/WebsiteEnvironment";
+import { ErrorMessage } from "../../components/ErrorMessage/ErrorMessage";
+import { Pagination } from "../../components/Pagination/Pagination";
+
+import styles from "./ServicesList.module.scss";
+
+type CardMetaData = {
+	label?: string;
+	value: React.ReactNode;
+};
+
+type ServicesListProps = {
+	basename: string;
+	location: {
+		pathname: string;
+		search: string;
+	};
+	history: HistoryType;
+};
+
+type ServicesListState = {
+	path: string;
+	originalWebsites: Array<WebsiteType>;
+	websites: Array<WebsiteType>;
+	searchQuery?: string;
+	error?: Error;
+	isLoading: boolean;
+	environmentFiltersChecked: Array<string>;
+	pageNumber: number;
+	itemsPerPage: number | string;
+	environmentsForFilter: Array<string>;
+};
+
+export class ServicesList extends Component<ServicesListProps, ServicesListState> {
+	constructor(props: ServicesListProps) {
+		super(props);
+
+		const querystring = this.props.location.search;
+
+		const querystringObject = queryStringToObject(querystring);
+
+		const pageNumber = Number(
+			querystringObject.page
+				? Array.isArray(querystringObject.page)
+					? querystringObject.page[0]
+					: querystringObject.page
+				: 1,
+		);
+
+		let itemsPerPage = querystringObject.amount
+			? Array.isArray(querystringObject.amount)
+				? querystringObject.amount[0]
+				: querystringObject.amount
+			: 25;
+
+		itemsPerPage = Number(itemsPerPage) ? Number(itemsPerPage) : itemsPerPage;
+
+		this.state = {
+			path: "",
+			originalWebsites: [],
+			websites: [],
+			isLoading: true,
+			pageNumber: pageNumber,
+			itemsPerPage: itemsPerPage,
+			environmentFiltersChecked: [],
+			environmentsForFilter: [],
+		};
+
+		document.title = "NICE Accounts - Services list";
+	}
+
+	async componentDidMount(): Promise<void> {
+		this.setState({ isLoading: true });
+
+		const websites = await fetchData(Endpoints.websitesList);
+
+		if (isDataError(websites)) {
+			this.setState({ error: websites });
+		}else{
+		const allEnvironments = websites.map((website: { environment: { name: string; }; }) => website.environment.name);
+
+		const environmentsForFilter = allEnvironments.reduce(function (accumulatedEnvironments: string[], currentEnvironment: string) {
+			if (accumulatedEnvironments.indexOf(currentEnvironment) === -1) {
+				accumulatedEnvironments.push(currentEnvironment)
+			}
+			return accumulatedEnvironments
+		}, []);
+
+		this.setState({environmentsForFilter})
+		}
+
+		this.setState({ originalWebsites: websites, websites: websites, isLoading: false });
```

Comment:
```suggestion
		this.setState({ originalWebsites: websites, websites, isLoading: false });
```

Reply:
Sorry these are probably down to preference really. Can leave them if you prefer.